src/plugins/admonition/schema.js

```diff
@@ -34,6 +34,14 @@ function normalizeAdmonition(change, error) {
         console.warn('Unhandled admonition violation:', code)
         break
 
+    case 'child_type_invalid':
+        if (child.type === 'figure_caption') {
+            change.unwrapBlockByKey(node.key)
+            return
+        }
```

Comment:
I don't understand what's the goal here. Why would you want to unwrap the entire note when it has a `figure_caption` child, but not when it has any other invalid child?

Reply:
This is specific case when user insert note in figure_caption. Other invalid nodes are just unwrapped.

Reply:
This normalization (`child_type_invalid`) is triggered when `admonition` contains an invalid child, not when it is an invalid child. It will not trigger when an admonition is a child of `figure_caption`.

Reply:
This normalization is triggered when `figure_caption` ins wrapped with `note` after `change.insertAdmonition('note')`

Reply:
Oh, I see. I'd prefer we avoid making normalizations for such very specific cases. I agree that this particular normalization might be the best thing to do in such cases, but it arguably should never have happened in the first place; the UI should've prevented the user from making such change, and if it's a result of another legal change, then it's a bug. To sum up my approach to normalizations: they exist to fix documents which became invalid during a course of normal editing, not to best solve pathological cases which should never have happened.